lib/segment/src/index/field_index/mmap_point_to_values.rs

```diff
@@ -0,0 +1,389 @@
+        let file_size = header_size + ranges_size + values_size;
+
+        // create new file amd mmap
+        let file_name = path.join(POINT_TO_VALUES_PATH);
+        create_and_ensure_length(&file_name, file_size)?;
+        let mut mmap = open_write_mmap(&file_name)?;
+
+        // fill mmap file data
+        let header = Header {
+            ranges_start: header_size as u64,
+            points_count: points_count as u64,
+        };
+        header
+            .write_to_prefix(mmap.as_mut())
+            .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))?;
+
+        // counter for values offset
+        let mut point_values_offset = header_size + ranges_size;
+        for (point_id, values) in iter {
+            let range = MmapRange {
+                start: point_values_offset as u64,
+                count: values.len() as u64,
+            };
+            range
+                .write_to_prefix(
+                    &mut mmap[header_size + point_id as usize * std::mem::size_of::<MmapRange>()..],
+                )
+                .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))?;
+
+            for value in values {
+                value.write_to_mmap(mmap.as_mut(), point_values_offset)?;
+                point_values_offset += value.mmaped_size();
+            }
+        }
+
+        let mmap = unsafe { MmapSlice::try_from(mmap)? };
+        mmap.flusher()()?;
+
+        Ok(Self {
+            file_name,
+            mmap,
+            header,
+        })
+    }
+
+    pub fn open(path: &Path) -> OperationResult<Self> {
+        let file_name = path.join(POINT_TO_VALUES_PATH);
+        let mmap = open_write_mmap(&file_name)?;
+        let header =
+            Header::read_from_prefix(mmap.as_ref()).ok_or(OperationError::InconsistentStorage {
+                description: NOT_ENOUGHT_BYTES_ERROR_MESSAGE.to_owned(),
+            })?;
+        let mmap = unsafe { MmapSlice::try_from(mmap)? };
+
+        Ok(Self {
+            file_name,
+            mmap,
+            header,
+        })
+    }
+
+    fn flusher(&self) -> Flusher {
+        self.mmap.flusher()
+    }
+
+    fn files(&self) -> Vec<PathBuf> {
+        vec![self.file_name.clone()]
+    }
+
+    pub fn get_values<'a, T: MmapValue<'a>>(
+        &'a self,
+        point_id: PointOffsetType,
+    ) -> OperationResult<Box<dyn Iterator<Item = OperationResult<T>> + 'a>> {
+        // first, get range of values for point
+        let range = if point_id < self.header.points_count as PointOffsetType {
+            let range_offset = (self.header.ranges_start as usize)
+                + (point_id as usize) * std::mem::size_of::<MmapRange>();
+            self.mmap
+                .get(range_offset..range_offset + std::mem::size_of::<MmapRange>())
+                .and_then(MmapRange::read_from)
+                .unwrap_or_default()
+        } else {
+            MmapRange { start: 0, count: 0 }
```

Comment:
We could do

```suggestion
            MmapRange::default()
```

though I'm fine with keeping it explicit.

Reply:
Thanks! Fixed. I use an empty iterator instead as the most explicit way to handle such points